synapse/federation/transport/server.py

```diff
@@ -286,7 +288,7 @@ def new_func(request, *args, **kwargs):
         return new_func
 
     def register(self, server):
-        pattern = re.compile("^" + PREFIX + self.PATH + "$")
+        pattern = re.compile("^" + self.PREFIX + self.PATH + "$")
```

Comment:
I'm unconvinced that:

```
    PATH = "/invite/(?P<context>[^/]*)/(?P<event_id>[^/]*)"
    PREFIX = FEDERATION_V2_PREFIX
```

is clearer than

```
    PATH = "/v2/invite/(?P<context>[^/]*)/(?P<event_id>[^/]*)"
```

(in other words: although it's a ballache, would it be better to go through adding `/v1` to the starts of all the `PATH`s?)